src/irmin-pack/inode.ml

```diff
@@ -482,7 +482,7 @@ struct
             1 + dynamic_size_of_v_encoding s !offref
         | Inode_v2_root | Inode_v2_nonroot ->
             let len = decode_bin_int s offref in
-            len - H.hash_size
+            len + 1 (*for encoding len*) + 1 (*for encoding magic*)
```

Comment:
The length can be encoded on more that 1 byte. 

I recommend this diff:

```diff
+        let magic_len = 1 in
         match kind with
         | Pack_value.Kind.Inode_v1_unstable | Inode_v1_stable ->
-            1 + dynamic_size_of_v_encoding s !offref
+            let vlen = dynamic_size_of_v_encoding s !offref in
+            magic_len + vlen
         | Inode_v2_root | Inode_v2_nonroot ->
-            let len = decode_bin_int s offref in
-            len + 1 (*for encoding len*) + 1 (*for encoding magic*)
+            let before = !offref in
+            let vlen = decode_bin_int s offref in
+            let after = !offref in
+            let lenlen = after - before in
+            magic_len + lenlen + vlen
```

which leads to this code:

```
	let magic_len = 1 in
        match kind with
        | Pack_value.Kind.Inode_v1_unstable | Inode_v1_stable ->
            let vlen = dynamic_size_of_v_encoding s !offref in
            magic_len + vlen
        | Inode_v2_root | Inode_v2_nonroot ->
            let before = !offref in
            let vlen = decode_bin_int s offref in
            let after = !offref in
	    let lenlen = after - before in
	    magic_len + lenlen + vlen
```

Reply:
Thanks, that indeed looks safer like this.